### DISCUSSION TITLE
clone axis x & y

### DISCUSSION THREAD

**Discussion by lgjean** [fra589/grbl-Mega-5X#204] [Q&A] [answered]
hello, I'm looking for a solution to clone the x and y axes for two other axes that need to move at the same time.
I am using an arduino mega with ramp1.4 with GRBL for a hot wire cutting machine.
https://www.zupimages.net/up/21/25/ith2.jpg
I would like to modify the code if possible with a modest level of programming. 
Thanks for your help !

**Comment by fra589** [comment, marked as answer]
Hi @lgjean,  
  
You don't have to modify the code to do what you want.
I understand your need is 4 linear axis with 2 axis X and 2 axis Y..
You just need to adjust the config.h file like this:  

```
#define N_AXIS 4            // Number of axes (3 to 6)
#define N_AXIS_LINEAR 4     // Number of linears axis

#define AXIS_1 0        // Axis indexing value. Must start with 0 and be continuous.
#define AXIS_1_NAME 'X' // Axis names must be in X, Y, Z, A, B, C, U, V, W, D, E & H.
#define AXIS_2 1
#define AXIS_2_NAME 'X'
#define AXIS_3 2
#define AXIS_3_NAME 'Y'

#if N_AXIS <3
  #error "N_AXIS must be >= 3. N_AXIS < 3 is not implemented."
#endif
#if N_AXIS > 3
  #define AXIS_4 3
  #define AXIS_4_NAME 'Y' // Letter of axis number 4
#endif
```
  
So you will connect:
- 1st X to axis number 1
- 2nd X to axis number 2
- 1st Y to axis number 3
- 2nd Y to axis number 4
See https://github.com/fra589/grbl-Mega-5X/wiki/grbl-Mega-5X-pinout for pinout diagram.

@++;
Gauthier.

**Reply by zalexzperez** [reply]
I think on 2nd Y you meant connect to axis number 4, not number 3. Right? 
That part got me a little confused, but after reading everything carefully along with the pinout page, I think I got it.

I'm planning on also controlling a foam cutter with **dual motors for X axis** and a **single motor for Y axis**, with a MKS GEN L board,  hotwire powered from the bed heating output. So that would make my cutter a 3-axis machine, right?

**Reply by fra589** [reply]
Hi @zalexzperez, You are right 2nd Y is connected to axis number 4! Thanks. Corrected.

**Comment by lgjean** [comment]
thank you for helping me !
I made the change to config.h and uploaded, then I changed the hookups as shown. I launched the UGS software and when I click on one of the two Y's in the software I just have X2 moving. When I click on one of the two Xs, I just have Y1 mov.
I want both X to move at the same time, and both Y to move at the same time.

**Comment by fra589** [comment]
Hi @lgjean,  
  
The behavior you describe is very strange...  
It happens sometime when Arduino's EEPROM is corrupted by a preceding axis configuration, Grbl can have a completely unpredictable behavior.  
When you change the axis number and / or name definition, you need to reset the configuration by issuing the `$RST=*` Grbl command.
And after, don't forget to give the same parameters to the cloned axis. In your case, $100 and $101 must have exactly the same value, and also : $102 and $103, $110 & $111, $112 & $113, $120 & $121, $122 & $123, $130 & $131 and finally $132 & $133.  
  
This should work. If it doesn't, I will ned more info to understand. So, you will need to post here the result of the 2 Grbl commands `$I` and `$$`.  
  
@++;  
Gauthier.

**Reply by lgjean** [reply]
hello, I disassembled my RAMPS and checked the welds, some were questionable which caused random behavior. I also had an error in the location of a driver, I had misunderstood that it was necessary to put one on Z. I then reset the eeprom and upload again with your modification and to now it works as desired! Thank you very much for your help and your patience, I close the topic.

**Comment by fra589** [comment]
Hi @lgjean,  
  
For a hot wire cutting machine, if you use axis cloning, you won't be able to do anything other than parts with parallel edges ...
If you want to be able to use it for example to make trapezoidal wings, you have to configure the right and left axes of your machine independently. So you need 4 linear axes X, Y, U and V.  
```
#define N_AXIS 4            // Number of axes (3 to 6)
#define N_AXIS_LINEAR 4     // Number of linears axis

#define AXIS_1 0        // Axis indexing value. Must start with 0 and be continuous.
#define AXIS_1_NAME 'X' // Axis names must be in X, Y, Z, A, B, C, U, V, W, D, E & H.
#define AXIS_2 1
#define AXIS_2_NAME 'Y'
#define AXIS_3 2
#define AXIS_3_NAME 'U'

#if N_AXIS <3
  #error "N_AXIS must be >= 3. N_AXIS < 3 is not implemented."
#endif
#if N_AXIS > 3
  #define AXIS_4 3
  #define AXIS_4_NAME 'V' // Letter of axis number 4
#endif
```
U being parallel to X and V parallel to Y.  
So, you will need to connect for the left side of your machine: X to the X output of RAMPS, Y to the Y output, and for the right side: U to the Z output and V to the E0 connector.  

Thus, you will be able to program the 2 driving sides of your hot wire independently with a single GCode command such as:
`G<0/1> X<valX> Y<valY> U<valU> V<valV>`  
  
This is normally how a hot wire cutting machine should work.  
  
@++;  
Gauthier.

**Comment by lgjean** [comment]
Thank you for the precision provided, if I clamp my machine in 2 axes, it is mainly because of the software which generates the GCODE to manage the 4 axes. This kind of application is quite expensive and I have to save money to buy it. In the meantime I'll just cut parallel objects which is enough to start having fun.

Jean

**Reply by zalexzperez** [reply]
How are you currently generating your designs? 
I'm using CorelDraw + eCut plugin which lets you connect each letter. And then I'm using Vectric Aspire to generate the gcode.
